### DISCUSSION TITLE
Backup data in nats-server with jetstream enabled where storage is memory

### DISCUSSION THREAD

**Discussion by anil1996** [nats-io/nats-server#4342] [Q&A] [answered]
Hello,

Is there a way we can take backup of streams and consumers for nats-server with jetstream enabled where storage is memory??
Or how to handle nats-server restart when storage is memory for jetstream?

**Comment by derekcollison** [comment, marked as answer]
Not currently. 

If you run the stream as an R3 and do rolling restarts and /healthz checks it will survive.

If you really need to, create a file backed mirror and back that up.

**Comment by jnmoyne** [comment]
If it's a controlled restart and your stream is R1 (i.e. replicas=1) you can do it seamlessly (without any interruption of service on the stream): before the restart update the stream's configuration to set replicas=3, check using `nats stream info` that all the new replicators have caught up, restart your server and then update the stream's configuration back to replicas=1.

If it's fault-tolerance you need (unscheduled server restart) then you must use replicas=3
